### PR TITLE
fix update issues

### PR DESCRIPTION
**Pull request by Berthalamew**
- fix skeleton issue in zombies
- fix map slots on dedis

### CODE REVIEW

File: xlive/Blam/Engine/game/players.cpp [hunk 1]
```diff
@@ -157,3 +157,132 @@ uint32 player_appearance_required_bits()
...
+{
+    // Campaign verification
+    if (game_is_campaign())
+    {
+        scenario* scnr = get_global_scenario();
+        uint32 block_size = scnr->player_starting_locations.size;
+        if (block_size > 0)
+        {
+            uint32 i = 0;
+            while (true)
```
Comment (nukeulater) [inline xlive/Blam/Engine/game/players.cpp]: Are you sure you're not missing any breaks here, in this loop?

Comment (nukeulater) [reply xlive/Blam/Engine/game/players.cpp]: Also why not a normal `for` loop?

File: xlive/Blam/Engine/game/players.cpp [hunk 2]
```diff
@@ -157,3 +157,132 @@ uint32 player_appearance_required_bits()
...
+    if (bungie_user_role < 0 || bungie_user_role > 7)
+    {
+        configuration_data->bungie_user_role = 0;
+    }
+
+    if (current_game_engine())
+    {
+        if (TEST_BIT(get_game_variant()->game_engine_flags, _game_engine_teams_bit))
+        {
+            if (configuration_data->team_index != NONE && (FLAG(configuration_data->team_index) & s_game_engine_globals::get()->flags_A) == 0)
```
Comment (nukeulater) [inline xlive/Blam/Engine/game/players.cpp]: `!TEST_BIT(s_game_engine_globals::get()->flags_A, configuration_data->team_index)`

**Review by nukeulater — CHANGES_REQUESTED** [review]

File: xlive/Blam/Engine/game/players.cpp (hunk 2 above)
Comment (Berthalamew) [reply xlive/Blam/Engine/game/players.cpp]: changed

File: xlive/Blam/Engine/game/players.cpp (hunk 1 above)
Comment (Berthalamew) [reply xlive/Blam/Engine/game/players.cpp]: Changed to a for loop.

**Pull request merged**